sdk/core/src/policies/retry_policies/retry_policy.rs

```diff
@@ -20,6 +20,13 @@ pub trait RetryPolicy {
     fn is_expired(&self, duration_since_start: Duration, retry_count: u32) -> bool;
     /// Determine how long before the next retry should be attempted.
     fn sleep_duration(&self, retry_count: u32) -> Duration;
+    /// A Future that will wait until the request can be retried.
+    fn wait(
+        &self,
+        retry_count: u32,
+    ) -> std::pin::Pin<std::boxed::Box<dyn std::future::Future<Output = ()> + Send>> {
```

Comment:
If we use the `async_trait` macro this would be much simpler.

Reply:
Sure, happy to do that. It only took me 10 min to figure our this signature so I could avoid using it 😭.